File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack=[]
        def judge(b,a):
            if (a=="(" and b==")") or (a=="[" and b=="]") or (a=="{"and b=="}"):
                return True
            else:return False
        
        for i in range(len(s)):
            stack.append(s[i])
            if len(stack)>1 and judge(stack[len(stack)-1],stack[len(stack)-2]):
                stack.pop()
                stack.pop()
        return not stack

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        vector<char> stack;
        for(int i=0;i<s.size();i++){
            if(s[i]=='('||s[i]=='['||s[i]=='{'){
                stack.push_back(s[i]);
                continue;
            }
            if(s[i]==')'){
                if(!stack.empty()&&stack.back()=='('){
                    stack.pop_back();
                    continue;
                }
                else{
                    return false;
                }
            }
            if(s[i]==']'){
                if(!stack.empty()&&stack.back()=='['){
                    stack.pop_back();
                    continue;
                }
                else{
                    return false;
                }
            }
            if(s[i]=='}'){
                if(!stack.empty()&&stack.back()=='{'){
                    stack.pop_back();
                    continue;
                }
                else{
                    return false;
                }
            }
        }
        if(stack.size()!=0){
            return false;
        }
        return true;
    }
};